ui/site/src/component/top-bar.ts

```diff
@@ -55,10 +55,8 @@ export default function () {
               if (!isVisible('#challenge-app')) $toggle.trigger('click');
             },
             setCount(nb: number) {
-              $countSpan
-                .attr('title', siteTrans('challengesX', nb))
-                .attr('aria-label', siteTrans('challengesX', nb))
-                .data('count', nb);
+              const newTitle = $countSpan.attr('title')!.replace(/\d+/, nb.toString());
+              $countSpan.data('count', nb).attr('title', newTitle).attr('aria-label', newTitle);
```

Comment:
I considered something like this before learning how to use the clientside portion of the translations system, but didn't like it because it felt a bit too uncertain whether any of the translations might include something matching `/\d+/`. I suppose that's very unlikely, still feels pretty hacky though.

Does the addition of translation keys in the main layout make much of an impact, or is it more that adding them there is a bit of a hacky workaround in itself?

Reply:
adding the translation keys in the layout is the *right* thing to do. But it does add some weight and parsing time to *every* homepage load. So here I decided that we could use a little hack instead.